mesonbuild/backend/xcodebackend.py

```diff
@@ -247,7 +247,7 @@ def object_filename_from_source(self, target, source):
         project = self.build.project_name
         buildtype = self.buildtype
         tname = target.get_id()
-        arch = 'x86_64'
+        arch = platform.machine().lower()
```

Comment:
Shouldn't this use the architecture from `self.environment.machines[target.for_machine].cpu_family`? It corresponds to the .cpu_family() method in `meson.build`, depending on whether the current target is `native: true` or not.

Reply:
This has been addressed in supporting cross-compiling.

Reply:
It sounds like this should all be one commit, then.